File: BeitPatissierServer/Models/RecipeLine.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BeitPatissierServer.Models
{
    public class RecipeLine
    {
        public int RecipeLineId { get; set; }
        public int RecipeId { get; set; }
        [ForeignKey(nameof(RecipeId))]
        public Recipe Recipe { get; set; }
        public int IngredientId { get; set; }
        [ForeignKey(nameof(IngredientId))]
        public Ingredient Ingredient { get; set; }
        public UnitEnum Unit { get; set; } // יחידת מידה (למשל: ק"ג, ליטר)

        public decimal Quantity { get; set; } // כמות חומר הגלם במתכון
    }

    public enum UnitEnum
    {
        [Display(Name = "קילו")]
        Kilo,
        [Display(Name = "גרם")]
        Gram,
        [Display(Name = "ליטר")]
        Liter,
        [Display(Name = "מלליטר")]
        Milliliter,
        [Display(Name = "כף")]
        Tablespoon,
        [Display(Name = "כפית")]
        Teaspoon,
        [Display(Name = "קורט")]
        Pinch,
        [Display(Name = "יחידה")]
        Unit
    }
}

File: beit-patissier-manage/src/models/RecipeLine.ts
import { Recipe } from './Recipe';
import { Ingredient } from './Ingredient';


export enum UnitEnum {
    Kilo = 'Kilo',
    Gram = 'Gram',
    Liter = 'Liter',
    Milliliter = 'Milliliter',
    Tablespoon = 'Tablespoon',
    Teaspoon = 'Teaspoon',
    Pinch = 'Pinch',
    Unit = 'Unit'
}

export class RecipeLine {
    recipeLineId: number;
    recipeId: number;
    recipe: Recipe;
    ingredientId: number;
    ingredient: Ingredient;
    unit: UnitEnum;
    quantity: number;

    
    constructor(
        recipeLineId: number,
        recipeId: number,
        recipe: Recipe,
        ingredientId: number,
        ingredient: Ingredient,
        unit: UnitEnum,
        quantity: number
    ) {
        this.recipeLineId = recipeLineId;
        this.recipeId = recipeId;
        this.recipe = recipe;
        this.ingredientId = ingredientId;
        this.ingredient = ingredient;
        this.unit = unit;
        this.quantity = quantity;
    }
}

export const UnitEnumHebrew: Record<UnitEnum, string> = {
    [UnitEnum.Kilo]: 'קילו',
    [UnitEnum.Gram]: 'גרם',
    [UnitEnum.Liter]: 'ליטר',
    [UnitEnum.Milliliter]: 'מיליליטר',
    [UnitEnum.Tablespoon]: 'כף',
    [UnitEnum.Teaspoon]: 'כפית',
    [UnitEnum.Pinch]: 'קמצוץ',
    [UnitEnum.Unit]: 'יחידה'
};